lib/luminork-server/src/routes.rs

```diff
@@ -0,0 +1,350 @@
+    "Reach out to support@systeminit.com ",
+    "or in the SI Discord for more information if this problem persists",
+);
+
+async fn app_state_middeware<B>(
+    State(state): State<AppState>,
+    request: Request<B>,
+    next: Next<B>,
+) -> Response {
+    match *state.application_runtime_mode.read().await {
+        ApplicationRuntimeMode::Maintenance => {
+            // Return a 503 when the server is in maintenance/offline
+            (StatusCode::SERVICE_UNAVAILABLE, MAINTENANCE_MODE_MESSAGE).into_response()
+        }
+        ApplicationRuntimeMode::Running => next.run(request).await,
+    }
+}
+
+// Define the OpenAPI document for your API
+#[derive(OpenApi)]
+#[openapi(
+    paths(
+        system_status_route,
+        crate::service::whoami::whoami,
+    ),
+    components(
+        schemas(
+            SystemStatusResponse,
+            ErrorResponse,
+            ErrorDetail,
+            crate::service::whoami::WhoamiResponse,
+            crate::service::v1::common::ApiSuccess<String>,
+            crate::service::v1::common::ApiError,
+        )
+    ),
+    tags(
+        (name = "root", description = "Root API endpoints"),
+        (name = "whoami", description = "User identity endpoints"),
+        (name = "change_sets", description = "Change set management endpoints"),
+        (name = "components", description = "Component management endpoints"),
+        (name = "management", description = "Management function endpoints")
+    ),
+    info(
+        title = "Luminork API",
+        description = "System Initiative External API server",
+        version = "1.0.0"
+    ),
+    servers(
+        (url = "/", description = "Public API Server for System Initiative")
+    )
+)]
+struct ApiDoc;
+
+// Define the schema for the system status response
+#[derive(serde::Serialize, ToSchema)]
+struct SystemStatusResponse {
+    #[schema(example = "I am luminork, the new System Initiative External API server")]
+    #[serde(rename = "What is this?")]
+    what_is_this: String,
+
+    #[schema(example = "Available at /swagger-ui")]
+    #[serde(rename = "API Documentation")]
+    api_documentation: String,
+}
+
+// Define the schema for error responses
+#[derive(serde::Serialize, ToSchema)]
+struct ErrorResponse {
+    error: ErrorDetail,
+}
+
+#[derive(serde::Serialize, ToSchema)]
+struct ErrorDetail {
+    message: String,
+    code: i32,
+    status_code: u16,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn routes(state: AppState) -> Router {
+    async fn openapi_handler() -> Result<Json<utoipa::openapi::OpenApi>, (StatusCode, String)> {
+        let mut openapi = ApiDoc::openapi();
+        let v1_openapi = v1::get_openapi();
+
+        for (path, path_item) in v1_openapi.paths.paths {
+            openapi.paths.paths.insert(path, path_item);
+        }
+
+        if let Some(openapi_components) = openapi.components.as_mut() {
+            if let Some(v1_components) = v1_openapi.components {
```

Comment:
What happens if this is "some, none" or "none, some"? Still okay to boot the server?